src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -352,8 +354,13 @@ COMMIT TRANSACTION
                     _resourceTypeToId = resourceTypeToId;
                     _resourceTypeIdToTypeName = resourceTypeIdToTypeName;
                     _searchParamUriToId = searchParamUriToId;
-                    _systemToId = systemToId;
-                    _quantityCodeToId = quantityCodeToId;
+
+                    _systemToId = new FhirMemoryCache<int>("systemToId");
+                    _systemToId.AddRange(systemToId);
```

Comment:
By loading all these rows first into a local concurrent dictionary, and then copying from that collection into the cache, we will still encounter problems if the system table has too many rows.  Could we add the values into the cache directly in the SQL read loop? And avoid the ConcurrentDictionary object?

Reply:
@feordin , you are more than correct!! Let me do that!